File: 02-programacion-basica/08-cadenas-y-caracteres/scala/ejemplos-guiados/CadenasYCaracteres.scala
object CadenasYCaracteres:
  def main(args: Array[String]): Unit =
    val entrada = "  Chilete   DevPath  "

    // split + filter + mkString normaliza espacios sin modificar la cadena original.
    val normalizada = entrada.trim.split("\\s+").filter(_.nonEmpty).mkString(" ").toUpperCase
    val vocales = normalizada.count(c => "AEIOU".contains(c))

    println(s"Texto: $normalizada")
    println(s"Vocales: $vocales")

File: 02-programacion-basica/08-cadenas-y-caracteres/java/ejemplos-guiados/CadenasYCaracteres.java
public class CadenasYCaracteres {
    public static void main(String[] args) {
        String entrada = "  Chilete   DevPath  ";

        // Cada operación devuelve una nueva cadena porque String es inmutable.
        String normalizada = entrada.trim().replaceAll("\\s+", " ").toUpperCase();
        int vocales = 0;

        for (char caracter : normalizada.toCharArray()) {
            if ("AEIOU".indexOf(caracter) >= 0) {
                vocales++;
            }
        }

        System.out.println("Texto: " + normalizada);
        System.out.println("Vocales: " + vocales);
    }
}
